docs/01-app/03-building-your-application/06-optimizing/14-local-development.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: Local Development
+description: Learn how to optimize your local development environment with Next.js.
+---
+
+Next.js is designed to provide a great developer experience. As your application grows, you might notice slower compilation times during local development. This guide will help you identify and fix common compile-time performance issues.
+
+## Local dev vs. production
+
+The development process with `next dev` is different than `next build` and `next start`.
+
+`next dev` compiles routes in your application as you open or navigate to them. This enables you to start the dev server without waiting for every route in your application to compile, which is both faster and uses less memory. Running a production build applies other optimizations, like minifying files and creating content hashes, which are not needed for local development.
```

Comment:
i think we can mention, this behaviour doesn't occur when we revisit the routes on subsequent visit since nextjs caches the compilation output.